core/src/main/scala/cats/data/IndexedReaderWriterStateT.scala

```diff
@@ -821,8 +836,54 @@ private trait IRWSTSemigroupK1[F[_], E, L, SA, SB] extends SemigroupK[IndexedRea
     }
 }
 
-private trait RWSTAlternative1[F[_], E, L, S]
+private trait RWSTNonEmptyAlternative1[F[_], E, L, S]
     extends IRWSTSemigroupK1[F, E, L, S, S]
+    with NonEmptyAlternative[ReaderWriterStateT[F, E, L, S, *]] {
+
+  implicit def F: Monad[F]
+  def G: NonEmptyAlternative[F]
+  implicit def L: Monoid[L]
+
+  // Enforces binary compatibility for RWSTAlternative1 between 2.6.1 and newer versions.
+  final protected def pure0[A](a: A): ReaderWriterStateT[F, E, L, S, A] =
+    ReaderWriterStateT.pure[F, E, L, S, A](a)
+
+  // Enforces binary compatibility for RWSTAlternative1 between 2.6.1 and newer versions.
+  final protected def ap0[A, B](
+    ff: ReaderWriterStateT[F, E, L, S, A => B]
+  )(
+    fa: ReaderWriterStateT[F, E, L, S, A]
+  ): ReaderWriterStateT[F, E, L, S, B] =
+    ff.flatMap(f => fa.map(f)(F))(F, L)
```

Comment:
Sorry, I don't really understand why this is necessary?

Reply:
Me either 😄  I mean, I don't understand why as well 😄

Reply:
fixed